ibc/modules/src/core/ics02_client/handler/update_client.rs

```diff
@@ -80,12 +80,16 @@ where
 	tracing::debug!("latest consensus state: {:?}", latest_consensus_state);
 
 	let now = ctx.host_timestamp();
-	let duration = now.duration_since(&latest_consensus_state.timestamp()).ok_or_else(|| {
+	let last_update_timestamp =
+		ctx.client_update_time(&client_id, client_state.latest_height()).map_err(|_| {
```

Comment:
is there any potential issue on updating the context before checking for expiration?

Reply:
We are not updating the context, that functions returns the timestamp when the last client update message was processed.